oxidized-transformers/src/ops/nonzero.rs

```diff
@@ -0,0 +1,221 @@
+use candle_core::{bail, CpuStorage, CustomOp1, Layout, Shape, Tensor, WithDType};
+use snafu::{ResultExt, Snafu};
+
+#[cfg(feature = "cuda")]
+use candle_core::CudaStorage;
+
+/// Non-zero indices errors.
+#[derive(Debug, Snafu)]
+pub enum NonzeroError {
+    #[snafu(display("Cannot apply nonzero op"))]
+    ApplyOp { source: candle_core::Error },
+
+    #[snafu(display("Failed to make tensor contiguous"))]
+    MakeContiguous { source: candle_core::Error },
+}
+
+/// Get indices of non-zero elements in a tensor.
+pub trait Nonzero {
+    fn nonzero(self) -> Result<Tensor, NonzeroError>;
+}
+
+impl Nonzero for &Tensor {
+    /// Get indices of non-zero elements in a tensor.
+    ///
+    /// Returns a tensor with shape *(n_nonzero, n_dim)* where each row
+    /// containt the index of a non-zero element. If the input is a 1D tensor,
+    /// a tensor with shape *(n_nonzero)* is returned.
+    fn nonzero(self) -> Result<Tensor, NonzeroError> {
+        let contiguous = self.contiguous().context(MakeContiguousSnafu)?;
+        contiguous.apply_op1(NonzeroOp).context(ApplyOpSnafu)
+    }
+}
+
+/// Non-zero indices op.
+///
+/// All op implementations will fail when the input tensor is not contiguous.
+/// The public `Nonzero::nonzero` method takes care of the conversion.
+struct NonzeroOp;
+
+impl NonzeroOp {
+    fn cpu_n_nonzero<T: WithDType>(
+        data: &[T],
+        layout: &Layout,
+    ) -> Result<usize, candle_core::Error> {
+        if !layout.is_contiguous() {
+            bail!("Nonzero op requires contiguous layout")
+        }
+
+        let n_nonzero = match layout.contiguous_offsets() {
+            Some((start_offset, end_offset)) => {
+                let data = &data[start_offset..end_offset];
+                data.iter().filter(|x| !x.is_zero()).count()
+            }
+            None => unimplemented!("Layout is contiguous"),
```

Comment:
```suggestion
            None => unimplemented!("Layout must be contiguous"),
```

Reply:
Since it this point the data is guaranteed to be contiguous, I replaced the whole `match` expression by an `unwrap` (like the earlier case you pointed out).